### PR TITLE
sys/fmt: add fmt_is_number()

### PR DESCRIPTION
**Pull request by haukepetersen**
### Contribution description
The `_is_number()` function is currently duplicated in the RIOT code base at two locations. Furthermore, I need the same function in some other module - and I really don't want to duplicate it even further... So I added that function to the `fmt` module.

So this PR does the following:
- move `fmt_is_digit()` and `fmt_is_upper()` to the `fmt`s modules public API
- add `fmt_is_number()` function to the `fmt` module
- add fitting unit tests for these 3 functions
- make `tests/driver_kw2xrf` and `shell/commands/sc_gnrc_netif` use `fmt_is_number()` instead of their local implementations

Note: while moving the `is_number()` function, I altered it slightly: (i) use `int` instead of `bool`, to keep the `fmt`s modules dependencies to the stdlib low, and (ii) also check the input string for being `NULL` so the function does not return `true` on empty strings.

### Testing procedure
Run unittests and check the build output for `tests/driver_kw2xrf` and `examples/gnrc_networking`.

### Issues/PRs references
none

### CODE REVIEW

File: tests/unittests/tests-fmt/tests-fmt.c [hunk 1]
```diff
@@ -20,6 +20,28 @@
 #include "fmt.h"
 #include "tests-fmt.h"
 
+static void test_fmg_is_x(void)
```
Comment (kaspar030) [inline tests/unittests/tests-fmt/tests-fmt.c]: typo here and below: `fmg` -> `fmt`

File: sys/fmt/fmt.c [hunk 2]
```diff
@@ -48,19 +48,23 @@ static const uint32_t _tenmap[] = {
...
-    return (c >= '0' && c <= '9');
+    return 'a' + (c - 'A');
 }
 
-static inline int _is_upper(char c)
+int fmt_is_number(const char *str)
 {
-    return (c >= 'A' && c <= 'Z');
-}
+    if (!str) {
```
Comment (kaspar030) [inline sys/fmt/fmt.c]:
```suggestion
    if (!str | !*str) {
```

otherwise an empty string would be considered a number.

Comment (haukepetersen) [reply sys/fmt/fmt.c]: make sense, fixed.

File: tests/unittests/tests-fmt/tests-fmt.c (hunk 1 above)
Comment (haukepetersen) [reply tests/unittests/tests-fmt/tests-fmt.c]: of course, fixed.

**Comment by haukepetersen** [comment]
addressed comments.

**Comment by kaspar030** [comment]
Sorry, tests were not run, I'll re-trigger.

**Comment by kaspar030** [comment]
ACK.

**Pull request merged**